endpoints/openrtb2/auction_test.go

```diff
@@ -1146,6 +1146,301 @@ func TestStoredRequests(t *testing.T) {
 	}
 }
 
+func TestMergeBidderParams(t *testing.T) {
```

Comment:
Added extensive unit tests for the merge methods. I left the json based integration tests alone.